### PR TITLE
Add descriptions and examples for `rawProc` and `rawEnv`

### PR DESCRIPTION
**Pull request by sls1005**
Add descriptions for `rawProc` and `rawEnv`. See <https://forum.nim-lang.org/t/10485> for more informations.

### CODE REVIEW

File: lib/system.nim [hunk 1]
```diff
@@ -2247,18 +2247,50 @@ when notJSnotNims:
   proc rawProc*[T: proc {.closure.} | iterator {.closure.}](x: T): pointer {.noSideEffect, inline.} =
     ## Retrieves the raw proc pointer of the closure `x`. This is
     ## useful for interfacing closures with C/C++, hash compuations, etc.
```
Comment (juancarlospaco) [inline lib/system.nim]:
```suggestion
    ## useful for interfacing closures with C/C++, hash computations, etc.
```

File: lib/system.nim [hunk 2]
```diff
@@ -2246,19 +2246,51 @@ when notJSnotNims:
...
+    ## This is not available for the JS target.
     #[
     The conversion from function pointer to `void*` is a tricky topic, but this
     should work at least for c++ >= c++11, e.g. for `dlsym` support.
     refs: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=57869,
     https://stackoverflow.com/questions/14125474/casts-between-pointer-to-function-and-pointer-to-object-in-c-and-c
     ]#
+    runnableExamples:
+      import std/sugar
+
```
Comment (Araq) [inline lib/system.nim]:
```suggestion
```

File: lib/system.nim [hunk 3]
```diff
@@ -2246,19 +2246,51 @@ when notJSnotNims:
...
+      proc makeClosure(x: int): (int -> int) =
```
Comment (Araq) [inline lib/system.nim]:
```suggestion
      proc makeClosure(x: int): (proc(y: int): int) =
```

File: lib/system.nim [hunk 4]
```diff
@@ -2246,19 +2246,49 @@ when notJSnotNims:
...
+        e = c1.rawEnv()
+        p = c1.rawProc()
+
+      if e.isNil():
+        let c2 = cast[proc(y: int): int {.nimcall.}](p)
+        echo c2(2)
+      else:
+        let c3 = cast[proc(y: int; env: pointer): int {.nimcall.}](p)
+        echo c3(3, e)
+
```
Comment (arnetheduck) [inline lib/system.nim:2284]: since we're here, these would be nice if they were implemented in pure nim or as magics - the emit here is backend-dependent

Comment (Araq) [reply lib/system.nim:2284]: Yes, but it should be a different PR.

**Pull request merged**